plugins/woocommerce/src/Internal/ReceiptRendering/Templates/order-receipt.php

```diff
@@ -3,99 +3,63 @@
-		table tr:last-child { color: #000000; font-weight: bold; }
-		footer {
-			font-size: <?php echo $constants['footer_font_size']; ?>pt;
-			border-top: 1px solid #707070;
-			margin-top: <?php echo $constants['margin']; ?>pt;
-			padding-top: <?php echo $constants['margin']; ?>pt;
-		}
-		p { line-height: <?php echo $constants['line_height']; ?>pt; margin: 0 0 <?php echo $constants['margin'] / 2; ?> 0; }
-		<?php if ( $payment_info ) { ?>
-			.card-icon {
-				width: <?php echo $constants['icon_width']; ?>pt;
-				height: <?php echo $constants['icon_height']; ?>pt;
-				vertical-align: top;
-				background-repeat: no-repeat;
-				background-position-y: center;
-				display: inline-block;
-				background-image: url("data:image/svg+xml;base64,<?php echo $payment_info['card_icon']; ?>");
-			}
-		<?php } ?>
+<?php echo $data['css']; ?>
 	</style>
 </head>
 
 <body>
 <header>
-	<h1><?php echo $texts['receipt_title']; ?></h1>
-	<h3><?php echo strtoupper( $texts['amount_paid_section_title'] ); ?></h3>
+	<h1 id="receipt_title"><?php echo $data['texts']['receipt_title']; ?></h1>
+	<h3 id="amount_paid_section_title"><?php echo strtoupper( $data['texts']['amount_paid_section_title'] ); ?></h3>
 	<p>
-		<?php echo $formatted_amount; ?>
+		<?php echo $data['formatted_amount']; ?>
 	</p>
-	<h3><?php echo strtoupper( $texts['date_paid_section_title'] ); ?></h3>
+	<h3 id="date_paid_section_title"><?php echo strtoupper( $data['texts']['date_paid_section_title'] ); ?></h3>
 	<p>
-		<?php echo $formatted_date; ?>
+		<?php echo $data['formatted_date']; ?>
 	</p>
-	<?php if ( $payment_method ) { ?>
-		<h3><?php echo strtoupper( $texts['payment_method_section_title'] ); ?></h3>
+	<?php if ( isset( $data['payment_method'] ) ) { ?>
+		<h3 id="payment_method_section_title"><?php echo strtoupper( $data['texts']['payment_method_section_title'] ); ?></h3>
 		<p>
-			<?php if ( $payment_info ) { ?>
-				<span class="card-icon"></span> - <?php echo $payment_info['card_last4']; ?>
+			<?php if ( $data['payment_info'] ) { ?>
+				<span class="card-icon"></span> - <?php echo $data['payment_info']['card_last4']; ?>
 			<?php } else { ?>
-				<p><?php echo $payment_method; ?></p>
+				<p><?php echo $data['payment_method']; ?></p>
 			<?php } ?>
 		</p>
 	<?php } ?>
 </header>
 
-<h3><?php echo strtoupper( $texts['summary_section_title'] ); ?></h3>
-<table>
+<h3 id="summary_section_title"><?php echo strtoupper( $data['texts']['summary_section_title'] ); ?></h3>
+<table id="line_items">
 	<?php
-	foreach ( $line_items as $line_item ) {
-		if ( isset( $line_item['quantity'] ) ) {
-			?>
-			<tr><td><?php echo $line_item['title']; ?> × <?php echo $line_item['quantity']; ?></td><td><?php echo $line_item['amount']; ?></td></tr>
-		<?php } else { ?>
-			<tr><td><?php echo $line_item['title']; ?></td><td><?php echo $line_item['amount']; ?></td></tr>
-			<?php
-		}
+	foreach ( $data['formatted_line_items'] as $formatted_line_item ) {
+		echo "<tr>$formatted_line_item</tr>";
```

Comment:
@Konamiman Can this `<tr>` element have a filter for a custom class? That's what Product Bundles, Composites, and Mix and Match all use to help style the resulting rows.

Reply:
You can use the `woocommerce_printable_order_receipt_css` filter to add custom CSS for these elements with `table#line_items tr`, would that help?

Reply:
I need to add rules to certain line item TRs but not all. For example a bundle line item will have different styling than the bundled line items. Traditionally we have a different class of the row for targeting. How could we do that without?

Reply:
Ok, makes sense. I've modified the `woocommerce_printable_order_receipt_formatted_line_item` filter so that now it includes the `<tr>` element too (so now the filter must return the entire table row).

Reply:
Is that preferable to adding a separate filter for just the class? This seem really hard if say both myself and another plugin need to look at the class as now we need to use regex/string matching to see if the `<tr>` already has a `class=` attribute.

Reply:
I have replaced the `woocommerce_printable_order_receipt_formatted_line_item` filter with a `woocommerce_printable_order_receipt_line_item_display_data` one that contains both the attributes for the `<tr>` element and the HTML that gets rendered inside it. I hope it's comprehensive enough now.